src/cpp/r/RUtil.cpp

```diff
@@ -159,6 +160,44 @@ core::Error iconvstr(const std::string& value,
    return Success();
 }
 
+std::set<std::string> makeRKeywords()
+{
+   std::set<std::string> keywords;
+   
+   keywords.insert("TRUE");
+   keywords.insert("FALSE");
+   keywords.insert("NA");
+   keywords.insert("NaN");
+   keywords.insert("NULL");
+   keywords.insert("NA_real_");
+   keywords.insert("NA_complex_");
+   keywords.insert("NA_integer_");
+   keywords.insert("NA_character_");
+   keywords.insert("Inf");
+   
+   keywords.insert("if");
+   keywords.insert("else");
+   keywords.insert("while");
+   keywords.insert("for");
+   keywords.insert("in");
+   keywords.insert("function");
+   keywords.insert("next");
+   keywords.insert("break");
+   keywords.insert("repeat");
+   keywords.insert("...");
+   
+   return keywords;
+}
+
+static const std::set<std::string> s_rKeywords = makeRKeywords();
```

Comment:
I've seen enough problems (especially across various versions of compilers) with construction/destruction of static C++ objects that I think we might be better off having both of these be function-level statics in `isRKeyword` (we can just do an empty check on s_rKeywords to see if it needs to be populated).